### DISCUSSION TITLE
how do i use "list_categories".

### DISCUSSION THREAD

**Discussion by makaloni6** [saleweaver/python-amazon-sp-api#190] [Q&A]
Hi.

I want to use "list_categories()".
My code is below.

・Code

```
res_catalog = Catalog(credentials=credentials, marketplace=Marketplaces.JP)
res = res_catalog.list_categories('B08MYJHQ84')
```

・result
`AttributeError: 'Catalog' object has no attribute 'list_categories'`

I checked this object by dir().
```
print(dir(res_catalog) )
-> 
['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_add_marketplaces', '_auth', '_check_response', '_request', '_request_a', '_request_grantless_operation', '_sign_request', 'auth', 'boto3_client', 'content_type', 'credentials', 'endpoint', 'get_item', 'grantless_auth', 'headers', 'list_items', 'marketplace_id', 'method', 'region', 'role', 'scheme', 'set_role', 'user_agent']
```

How do I use this method ?
Thank you.

**Comment by miogarun** [comment]
You can't use it now if you install library via pip because this function has not yet been added to the last release.

**Reply by Shahin-rmz** [reply]
how should we install it then?

**Comment by denisneuf** [comment]
As @miogarun mention if you do not update the catalog.py trough the last commit:
[https://github.com/saleweaver/python-amazon-sp-api/blob/master/sp_api/api/catalog/catalog.py](https://github.com/saleweaver/python-amazon-sp-api/blob/master/sp_api/api/catalog/catalog.py)
You cannot use that endpoint that I added, but when you update the sub-api you need pass your ASIN or SellerSKU as key argument so the right way should be:

```
    try:
        res = Catalog(credentials=credentials, marketplace=Marketplaces.JP).list_categories(
            ASIN='B08MYJHQ84',
        )
        logging.info(res)
    except SellingApiException as ae:
        logging.error(ae)
```

**Reply by saleweaver** [reply]
I'll add the updates in the next release, I'll publish on the weekend.

**Reply by makaloni6** [reply]
Thank you for replying!
I will try after update.
